src/streamer.rs

```diff
@@ -445,63 +445,70 @@ fn broadcast(
     while let Ok(mut nq) = r.try_recv() {
         dq.append(&mut nq);
     }
-    let mut blobs: Vec<_> = dq.into_iter().collect();
+    let blobs_vec: Vec<_> = dq.into_iter().collect();
+    let blobs_chunked: Vec<_> = blobs_vec
+        .chunks(window.read().unwrap().len())
+        .map(|x| x.to_vec())
+        .collect();
 
     print_window(window, *receive_index as usize);
 
-    // Insert the coding blobs into the blob stream
-    #[cfg(feature = "erasure")]
-    erasure::add_coding_blobs(recycler, &mut blobs, *receive_index);
+    for mut blobs in blobs_chunked {
+        // Insert the coding blobs into the blob stream
+        #[cfg(feature = "erasure")]
+        erasure::add_coding_blobs(recycler, &mut blobs, *receive_index);
 
-    let blobs_len = blobs.len();
-    info!("broadcast blobs.len: {}", blobs_len);
+        let blobs_len = blobs.len();
+        debug!("broadcast blobs.len: {}", blobs_len);
 
-    // Index the blobs
-    Crdt::index_blobs(crdt, &blobs, receive_index)?;
-    // keep the cache of blobs that are broadcast
-    {
-        let mut win = window.write().unwrap();
-        for b in &blobs {
-            let ix = b.read().unwrap().get_index().expect("blob index");
-            let pos = (ix as usize) % WINDOW_SIZE;
-            if let Some(x) = &win[pos] {
-                trace!(
-                    "popped {} at {}",
-                    x.read().unwrap().get_index().unwrap(),
-                    pos
-                );
-                recycler.recycle(x.clone());
+        // Index the blobs
+        Crdt::index_blobs(crdt, &blobs, receive_index)?;
+        // keep the cache of blobs that are broadcast
+        {
+            let mut win = window.write().unwrap();
+            assert!(blobs.len() <= win.len());
+            for b in &blobs {
+                let ix = b.read().unwrap().get_index().expect("blob index");
+                let pos = (ix as usize) % WINDOW_SIZE;
+                if let Some(x) = &win[pos] {
+                    trace!(
+                        "popped {} at {}",
+                        x.read().unwrap().get_index().unwrap(),
+                        pos
+                    );
+                    recycler.recycle(x.clone());
```

Comment:
Strange place for a `clone()`. Is that necessary?

Reply:
from @aeyakovenko yes, it's necessary. It's an element in the window array, so cannot be moved out..?

Reply:
```
warning: build failed, waiting for other jobs to finish...
error[E0507]: cannot move out of borrowed content
   --> src/streamer.rs:484:38
    |
484 |                     recycler.recycle(*x);
    |                                      ^^ cannot move out of borrowed content
```

Reply:
#451